packages/core/src/external/commonwell/document/document-file-type-detector.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * The function `detectFileType` uses magic numbers to determine the file type of a given file.
+ * Magic numbers are unique sequences of bytes that identify the file format or protocol.
+ * In this case, the function reads the first 5 bytes (the magic number) of the file buffer and
+ * matches it against known file type headers.
+ *
+ * @param {Buffer} fileBuffer - The `fileBuffer` parameter is a `Buffer` object that represents the
+ * contents of a file. The first 5 bytes of this buffer, which form the magic number, are used to
+ * identify the file type.
+ * @returns The function `detectFileType` returns a string representing the detected file type.
+ */
+export function detectFileType(fileBuffer: Buffer): string {
+  if (
+    (fileBuffer[0] === 0x49 &&
+      fileBuffer[1] === 0x49 &&
+      fileBuffer[2] === 0x2a &&
+      fileBuffer[3] === 0x00) ||
+    (fileBuffer[0] === 0x4d &&
+      fileBuffer[1] === 0x4d &&
+      fileBuffer[2] === 0x00 &&
+      fileBuffer[3] === 0x2a)
+  ) {
+    return "image/tiff";
+  } else if (
+    fileBuffer[0] === 0x25 &&
+    fileBuffer[1] === 0x50 &&
+    fileBuffer[2] === 0x44 &&
+    fileBuffer[3] === 0x46 &&
+    fileBuffer[4] === 0x2d
+  ) {
+    return "application/pdf";
+  } else if (
+    fileBuffer[0] === 0x3c &&
+    fileBuffer[1] === 0x3f &&
+    fileBuffer[2] === 0x78 &&
+    fileBuffer[3] === 0x6d &&
+    fileBuffer[4] === 0x6c &&
+    fileBuffer[5] === 0x20
+  ) {
+    return "application/xml";
+  } else if (
+    fileBuffer[0] === 0x89 &&
+    fileBuffer[1] === 0x50 &&
+    fileBuffer[2] === 0x4e &&
+    fileBuffer[3] === 0x47
+  ) {
+    return "image/png";
+  } else if (fileBuffer[0] === 0xff && fileBuffer[1] === 0xd8 && fileBuffer[2] === 0xff) {
+    return "image/jpeg";
+  } else {
+    throw new Error(`Unknown file type. Cannot convert document: ${fileBuffer.slice(0, 10)}`);
```

Comment:
throwing an error if we cant figure out the file type

Reply:
why are the bytes in the error message? wouldn't it be better to throw the doc ID or something for debugging?

also `Cannot convert document` isn't necessarily accurate, perhaps a better message could be `Could not determine file type for document: ${doc.id}`?